docs/contributing.rst

```diff
@@ -211,6 +211,13 @@ specify either as a default, in the :file:`.pypirc` file, or pass on
 the command line), and the methods that upload the package securely to
 a URL.
 
+For more details, refer to the source documentation (currently a
+`work in progress <https://github.com/pypa/twine/issues/635>`_):
+
+.. toctree::
+
+   internal/twine
+
```

Comment:
@sigmavirus24 Following up on https://github.com/pypa/twine/pull/810#discussion_r707234204, how does this plus the [top-level admonition](https://twine--810.org.readthedocs.build/en/810/internal/twine.html) look?

If you think this is worth merging, I'm inclined to rewrite this "Architectural Overview" section to link to the appropriate docs, though I'd probably do that in a subsequent PR.